src/utils.sh

```diff
@@ -52,23 +56,25 @@ CORRELATION_VECTOR=""
 ######################################
 
 function set_opt_out_selection() {
-    if [ $1 == true ];
+    if [[ "$LOCAL_E2E" == "1" || $1 == true ]];
     then
         OPT_IN=false
         log_info "The user has opted out of sending usage telemetry."
     else
         OPT_IN=true
         log_info "The user has opted in for sending usage telemetry."
-    fi
 
-    # handle correlation vector
-    if [ -z $2 ];
-    then
-        CORRELATION_VECTOR=$(generate_uuid)
-    else
-        CORRELATION_VECTOR=$2
-    fi
+        # handle correlation vector
+        if [ -z $2 ];
+        then
+            CORRELATION_VECTOR=$(generate_uuid)
+        else
+            CORRELATION_VECTOR=$2
+        fi
 
+        local ret_value=$(openssl dgst -hmac $3 <<< `echo $3$4`)
```

Comment:
>$3 [](http://example.com/codeflow?start=45&length=2)

can we pick a value that is not a secret here? We could just use a hardcoded GUID maybe? This way if the hashing is undone the plain value will not be identifiable.

Reply:
would using InstrumentationKey suffice?